python-lib/warp10/warp10client.py

```diff
@@ -0,0 +1,231 @@
+import logging
+import requests
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO,
+                    format='Warp10 client %(levelname)s - %(message)s')
+
+class WarpScriptExecutionError(Exception):
+    pass
+
+
+class FetchModeParameters:
+    def __init__(self):
+        self.fetch_mode = None
+        self.start = None
+        self.stop = None
+        self.now = None
+        self.timespan = None
+
+    def as_interval(self, start, stop):
+        self.fetch_mode = 'interval'
+        self.start = start
+        self.stop = stop
+
+    def as_timespan(self, now, timespan):
+        self.fetch_mode = 'timespan'
+        self.now = now
+        self.timespan = timespan
+
+    def is_interval(self):
+        return self.fetch_mode == 'interval'
+
+    def is_timespan(self):
+        return self.fetch_mode == 'timespan'
+
+
+class GTSStringBuilder:
+    GTS_INPUT_FORMAT = '{timestamp}/{latlon}/{elevation} {identifier} {value}'
+
+    def __init__(self):
+        self.timestamp = None
+        self.latitude = None
+        self.longitude = None
+        self.elevation = ''
+        self.identifier = None
+        self.value = None
+        self.value_type = None
+
+    def with_timestamp(self, timestamp):
+        self.timestamp = timestamp
+
+    def with_latitude(self, latitude):
+        self.latitude = latitude
+
+    def with_longitude(self, longitude):
+        self.longitude = longitude
+
+    def with_elevation(self, elevation):
+        self.elevation = elevation
+
+    def with_identifier(self, identifier):
+        self.identifier = identifier
+
+    def with_value(self, value, value_type):
+        self.value = value
+        self.value_type = value_type
+
+    def to_string(self):
+        # Better to throw if something is missing, or just log and skip this row?
+        if not self.identifier:
+            raise TypeError('Row is missing an identifier')
+        if not self.timestamp:
+            raise TypeError('Row is missing a timestamp')
+        if not self.value:
+            raise TypeError('Row is missing a value')
+        
+        latlon = ''
+        if self.latitude and self.longitude:
+            latlon = str(self.latitude) + ':' + str(self.longitude)
+
+        if not ('{' in self.identifier and '}' in self.identifier):  # Very dirty check for x{y}
+            self.identifier = self.identifier + '{}'
+
+        if self.value_type == 'string':
+            self.value = '"' + self.value + '"'
+
+        return self.GTS_INPUT_FORMAT.format(timestamp=self.timestamp, latlon=latlon, elevation=self.elevation, 
+                                            identifier=self.identifier, value=self.value)
+
+
+class Warp10Client:
+    ALL_SELECTOR = '~.*{}'
+    MAX_LONG = 9223372036854775807
+
+    def __init__(self, warp10_connection):
+        self.base_url = 'http://' + warp10_connection['host'] + ':' + str(int(warp10_connection['port'])) + '/api/v0/'
```

Comment:
Good catch, completely forgot about that. Either I add a checkbox to use https or not, or instead of `host` and `port`, let the user enter the url, like `https://warp10host:warp10port`

Reply:
Went with the `https://warp10host:warp10port` approach since we already do something similar for API nodes (7ed3ac3)